File: python/434.number-of-segments-in-a-string.py
class Solution:
    def countSegments(self, s: str) -> int:
        seg = 0
        prev = " "

        # count number of transitions from space to non-space
        for c in s:
            if c != " " and prev == " ":
                seg += 1
            prev = c

        return seg

File: cpp/434.number-of-segments-in-a-string.cpp
#include <string>

using std::string;

class Solution {
public:
  int countSegments(string s) {
    int seg = 0;
    char p = ' ';

    // count number of transitions from space to non-space
    for (char c : s) {
      if (p == ' ' && c != ' ') {
        seg++;
      }

      p = c;
    }

    return seg;
  }
};